### PR TITLE
[util] reoganize the device and add timeit

### PR DESCRIPTION
**Pull request by HuaizhengZhang**
1. reorganize the device placement
2. add timeit to check system performance

### CODE REVIEW

File: server/model_server/audio_model_server.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def load_soundnet():
 class Api2MslServicer(api2msl_pb2_grpc.Api2MslServicer):
     def __init__(self):
         super().__init__()
-        os.environ['CUDA_VISIBLE_DEVICES'] = '0'
+        os.environ['CUDA_VISIBLE_DEVICES'] = '2'
```
Comment (YuanmingLeee) [inline server/model_server/audio_model_server.py:52]: Could use a sperate .yml file hardware for configuration

File: server/model_server/product_search_model_server.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def load_search_machine():
 # Custom request servicer
 class Api2MslServicer(api2msl_pb2_grpc.Api2MslServicer):
     def __init__(self):
-        os.environ['CUDA_VISIBLE_DEVICES'] = '0'
+        os.environ['CUDA_VISIBLE_DEVICES'] = '2'
```
Comment (YuanmingLeee) [inline server/model_server/product_search_model_server.py:36]: Same as above

File: server/model_server/utils.py [hunk 3]
```diff
@@ -10,3 +10,25 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         sys.stdout.close()
         sys.stdout = self._original_stdout
+
+# https://medium.com/pythonhive/python-decorator-to-measure-the-execution-time-of-methods-fa04cb6bb36d
+def timeit(method):
```
Comment (YuanmingLeee) [inline server/model_server/utils.py]: Can we put this util function in `hysia.utils` as a general utility function?

File: server/model_server/audio_model_server.py (hunk 1 above)
Comment (YuanmingLeee) [reply server/model_server/audio_model_server.py:52]: E.g:
```yaml
decoder: CPU
model_server:
  audio: CUDA:2
  face: CUDA:0
  text: CUDA:0
  scene: CUDA:1
```

File: server/model_server/utils.py [hunk 4]
```diff
@@ -10,3 +10,25 @@ def __enter__(self):
...
+    """
+    def timed(*args, **kw):
+        ts = time.time()
+        result = method(*args, **kw)
+        te = time.time()
+        if 'log_time' in kw:
+            name = kw.get('log_name', method.__name__.upper())
+            kw['log_time'][name] = int((te - ts) * 1000)
+        else:
+            print('%r  %2.2f ms' % (method.__name__, (te - ts) * 1000))
```
Comment (YuanmingLeee) [inline server/model_server/utils.py]: Can we use a Logger? I believe there are some other outputs when running functions (e.g. gRPC call), use a Logger to output the result directly into a file would be nicer.

Comment (YuanmingLeee) [reply server/model_server/utils.py]: Also use the new format style would be better.
```suggestion
            print(f'{method.__name__!r}  {(te - ts) * 1000:2.2f} ms')
```

Comment (YuanmingLeee) [reply server/model_server/utils.py]:
```python
def timeit(method):
    """Use to time a running time of a method.
    Adapted from https://medium.com/pythonhive/python-decorator-to-measure-the-execution-time-of-methods-fa04cb6bb36d
    """
    def timed(*args, **kwargs):
        # obtain logger
        _log_time = kwargs.pop('_log_time', None)
        _log_name = kwargs.pop('_log_name', method.__name__)
        
        ts = time.time()
        result = method(*args, **kwargs)
        te = time.time()
        if _log_time is not None:
            _log_time[_log_name] = (te - ts) * 1000
        else:
            print(f'{method.__name__!r}  {(te - ts) * 1000:2.2f} ms')
        return result
    return timed
```

File: server/model_server/audio_model_server.py (hunk 1 above)
Comment (HuaizhengZhang) [reply server/model_server/audio_model_server.py:52]: add a device_placement.yml file for recording this information
add todo to use this file

We can leave this task later and now focus on testing the system manully.

File: server/model_server/product_search_model_server.py (hunk 2 above)
Comment (HuaizhengZhang) [reply server/model_server/product_search_model_server.py:36]: see new commits

File: server/model_server/utils.py (hunk 3 above)
Comment (HuaizhengZhang) [reply server/model_server/utils.py]: Done

Move the file to `hysia.utils.perf`

**Review by YuanmingLeee — APPROVED** [review]

**Pull request merged**